samples/rtl/axpy_double_pump.py

```diff
@@ -71,412 +71,361 @@
+    read_x_out = state.add_write('x_stream')
+    state.add_memlet_path(read_x_inp, read_x_entry, read_x, dst_conn='inp', memlet=dace.Memlet('fpga_x[i]'))
+    state.add_memlet_path(read_x, read_x_exit, read_x_out, src_conn='out', memlet=dace.Memlet('x_stream[0]'))
+
+    read_y_inp = state.add_read('fpga_y')
+    read_y_out = state.add_write('y_stream')
+    state.add_memlet_path(read_y_inp, read_y_entry, read_y, dst_conn='inp', memlet=dace.Memlet('fpga_y[i]'))
+    state.add_memlet_path(read_y, read_y_exit, read_y_out, src_conn='out', memlet=dace.Memlet('y_stream[0]'))
+
+    # add tasklet memlets
+    a = state.add_read('a')
+    x = state.add_read('x_stream')
+    y = state.add_read('y_stream')
+    result = state.add_write('result_stream')
+    state.add_memlet_path(a, rtl_tasklet, dst_conn='a_in', memlet=dace.Memlet('a[0]'))
+    state.add_memlet_path(x, rtl_tasklet, dst_conn='x_in', memlet=dace.Memlet('x_stream[0]'))
+    state.add_memlet_path(y, rtl_tasklet, dst_conn='y_in', memlet=dace.Memlet('y_stream[0]'))
+    state.add_memlet_path(rtl_tasklet, result, src_conn='result_out', memlet=dace.Memlet('result_stream[0]'))
+
+    # add write_c memlets and access nodes
+    write_result_inp = state.add_read('result_stream')
+    write_result_out = state.add_write('fpga_result')
+    state.add_memlet_path(write_result_inp,
+                        write_result_entry,
+                        write_result,
+                        dst_conn='inp',
+                        memlet=dace.Memlet('result_stream[0]'))
+    state.add_memlet_path(write_result,
+                        write_result_exit,
+                        write_result_out,
+                        src_conn='out',
+                        memlet=dace.Memlet('fpga_result[i]'))
+
+    # add copy to device state
+    copy_to_device = sdfg.add_state('copy_to_device')
+    cpu_x = copy_to_device.add_read('x')
+    cpu_y = copy_to_device.add_read('y')
+    dev_x = copy_to_device.add_write('fpga_x')
+    dev_y = copy_to_device.add_write('fpga_y')
+    copy_to_device.add_memlet_path(cpu_x, dev_x, memlet=dace.Memlet('x[0:N//VECLEN]'))
+    copy_to_device.add_memlet_path(cpu_y, dev_y, memlet=dace.Memlet('y[0:N//VECLEN]'))
+    sdfg.add_edge(copy_to_device, state, dace.InterstateEdge())
+
+    # add copy to host state
+    copy_to_host = sdfg.add_state('copy_to_host')
+    dev_result = copy_to_host.add_read('fpga_result')
+    cpu_result = copy_to_host.add_write('result')
+    copy_to_host.add_memlet_path(dev_result, cpu_result, memlet=dace.Memlet('result[0:N//VECLEN]'))
+    sdfg.add_edge(state, copy_to_host, dace.InterstateEdge())
+
+    # validate sdfg
+    sdfg.validate()
+
+    return sdfg
 
 ######################################################################
 
 if __name__ == '__main__':
+    # Configuration has to use multiple clocks in this sample, and target hardware.
+    dace.config.Config.set('compiler', 'xilinx', 'frequency', value='"0:300\\|1:600"')
```

Comment:
I would reset these changes (there is something for doing this in dace.config)

Reply:
Manually, like the new version is, or is there a specific function for doing this?

Reply:
There is a function to be used with `with` (`set_temporary`).
If that does not work, then manually

Reply:
I can't seem to find it anywhere, but it sounds like a more elegant solution! Do you have an example? Otherwise, it's been implemented manually.